CMakeLists.txt

```diff
@@ -21,8 +21,16 @@ option(SNMALLOC_OPTIMISE_FOR_CURRENT_MACHINE "Compile for current machine archit
 set(CACHE_FRIENDLY_OFFSET OFF CACHE STRING "Base offset to place linked-list nodes.")
 set(SNMALLOC_STATIC_LIBRARY_PREFIX "sn_" CACHE STRING "Static library function prefix")
 
+# malloc.h was removed from C in C89 and so FreeBSD's version has a #error and
```

Comment:
Not sure the history lesson is necessary.  Non-standard features aren't in the standard, so putting them in "malloc.h" seems perfectly reasonable.

Reply:
I am happy with the test fix, just not the history lesson in the comments.  The standard does not require the compiler to fail if you include "malloc.h" in C89 or later. I think FreeBSDs choice is a perfectly valid one, but it doesn't need documenting in detail here.

While fixing you might want to fix Apple as well, which uses
```C++
#include <malloc/malloc.h>
```
The file [external_alloc.h](https://github.com/microsoft/snmalloc/blob/master/src/mem/external_alloc.h) goes through various ways of including `malloc_usable_size`.

Reply:
I don't really want to lose the comment because, until we get FreeBSD in CI, someone is likely to look at the code say 'oh, you can simplify this by unconditionally including `<malloc.h>`' or 'We should just include `<malloc.h>` if it exists and see that CI is happy.

Reply:
```c++
# `malloc_usable_size` is an extension to the C runtime.
# This test code needs to find the relevant header. 
# Below we list the platforms and the relevant header: 
#   Linux            malloc.h
#   FreeBSD       malloc_np.h
#   Apple           malloc/malloc.h
```
We don't need to say why a particular platform made their choice the way they did.  It does not affect `snmalloc` functionality.

Reply:
I realised Apple has `malloc_size` so what I said there is incorrect.  

We could add a comment like:
```
# Including malloc.h is an error on FreeBSD.
```

Reply:
Thank you for the change the new comment states clearly what an snmalloc dev needs to know.